File: java/src/deepspace/SpaceCity.java
/*
 * Here comes the text of your license
 * Each line should be prefixed with  * 
 */
package deepspace;

import java.util.ArrayList;
/**
 *
 * @author yabir
 */
public class SpaceCity extends SpaceStation{
    
    private SpaceStation base;
    private ArrayList<SpaceStation> rest;
    
    SpaceCity(SpaceStation b, ArrayList<SpaceStation> rest){
        super(b);
        this.rest = rest;
        this.base = b;
    }
    
    ArrayList<SpaceStation> getCollaborators(){
        return rest;
    }
    
    @Override
    public float fire(){
        float factor = super.fire();
        for(SpaceStation station: getCollaborators()){
            factor += station.fire();
        }
        
        return factor;
    }
    
    @Override
    public float protection(){
        float factor = super.protection();
        
        for(SpaceStation station: getCollaborators()){
            factor += station.protection();
        }
        
        return factor;
    }
    
    @Override
    public Transformation setLoot(Loot l){
        super.setLoot(l);
        return Transformation.NOTRANSFORM;
    }
    
    @Override
    public SpaceCityToUI getUIversion(){
        return new SpaceCityToUI(this);
    }
    
}

File: rubyPreExamen/lib/SpaceCity.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'SpaceCityToUI'

module Deepspace
class SpaceCity < SpaceStation
  
  attr_reader :collaborators
  
  def initialize(base, rest)
    super(base.name, base.supplies)
    @base = base
    @collaborators = rest
  end
  
  def self.newCopy(station)
    h = new(station.base, station.collaborators)
  end
  
  def fire
    shot = super
    
    @collaborators.each{|station|
      shot *= station.fire
    }
    
    shot
  end
  
  def protection
    prot = super
    
    @collaborators.each{|station|
      prot += station.protection
    }
    
    prot
  end
  
  def setLoot(loot)
    super
    return Transformation::NOTRANSFORM
  end
  
  def getUIversion
    return SpaceCityToUI.new(self)
  end
end
end